mediapipe/tasks/ios/genai/core/sources/LlmTaskRunnerRefactored.swift

```diff
@@ -0,0 +1,88 @@
+// Copyright 2024 The MediaPipe Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import MediaPipeTasksGenAIC
+
+/// This class is used to create and call appropriate methods on the C `LlmInferenceEngine` to
+/// initialize an LLMInference task and create prompt sessions.
+final class LlmTaskRunner {
+  typealias CLlmEngine = UnsafeMutableRawPointer
+
+  /// The underlying C LLM engine created and managed by this `LlmTaskRunner`.
+  private var cLlmEngine: CLlmEngine?
+
+  /// Creates a new instance of `LlmTaskRunner` with the given model settings.
+  ///
+  /// - Parameters:
+  ///   - modelSettings: C model settings of type `LlmModelSettings`.
+  /// - Throws: An error if the engine could not be initialized.
+  init(modelSettings: LlmModelSettings) throws {
+    var cErrorMessage: UnsafeMutablePointer<CChar>? = nil
+    guard
+      (withUnsafePointer(to: modelSettings) {
+        LlmInferenceEngine_CreateEngine($0, &self.cLlmEngine, &cErrorMessage)
+      }) == StatusCode.success.rawValue
+    else {
+      throw GenAiInferenceError.failedToInitializeEngine(
+        String(allocatedCErrorMessage: cErrorMessage))
+    }
+  }
+
+  /// Creates a new C LLM session from the current C engine and returns an `LlmSessionRunner` 
+  /// that wraps around the newly created C session. The session runner is responsible for managing 
+  /// its underlying C session.
```

Comment:
Shall we mention that this method will always return a new instance even with the same `LlmSessionConfig`?

Reply:
We should. I have added a note in the comments.